de.fu_berlin.inf.dpp/build.gradle

```diff
@@ -0,0 +1,145 @@
+configurations {
+  stfTest
+}
+
+sourceSets {
+  main.java.srcDirs = ['src',
+                       'ext-src',
+                       'test/framework/stf/src',
+                       'test/framework/stf/test',
+                       'test/junit',
+                       'test/stf']
+
+  main {
+    resources {
+      srcDir 'src'
+      exclude '**/*.java'
+    }
+    resources {
+      srcDir 'test/framework/stf/src'
+      exclude '**/*.java'
+    }
+  }
+
+  test {
+    java.srcDir 'test/junit'
+    resources.srcDir 'test/resources'
+  }
+
+  stfTest {
+    java.srcDir 'test/stf'
+    resources.srcDir 'test/resources'
+  }
+}
+
+dependencies {
+  compile project(':de.fu_berlin.inf.dpp.core')
+  compile project(':de.fu_berlin.inf.dpp.ui')
+
+  compile project(path: ':de.fu_berlin.inf.dpp.core', configuration: 'testing')
+  compile project(path: ':de.fu_berlin.inf.dpp.ui', configuration: 'testing')
+
+  compile fileTree(dir: eclipsePluginDir).matching {
```

Comment:
So what are about OSGi dependencies. Are they correctly resolved or does this command simply adds jar files to the classpath for compiling ?

Remember:

https://www.infoq.com/news/2011/11/bndtools-11

InfoQ: How does BndTools compare to Eclipse's PDE?

    Neil Bartlett: Eclipse PDE is another OSGi development environment based on Eclipse. Bndtools does not reuse PDE in any way, but both Bndtools and PDE are built on top of Eclipse's Java Development Tooling (JDT).

    PDE follows a different philosophy to bnd and Bndtools, called "manifest first". In PDE you directly edit the MANIFEST.MF file that goes straight into the bundle without any post-processing.

To make it clear, and that is what Ant4Eclipse does. I choose the dependencies (contained in the Manifest.MF file) and it is the task of the build tool to verify that all dependencies are correctly resolved for compilation and that all classes are accessible (so this what OSGi is all about, module management). I do not really know if it is possible to tell Gradle that I want component XYZ in version range from 1.0.0 up to 3.4.1, or just any version, etc.

Reply:
You are right the current approach just adds all jar (except the excluded jars) as compile dependency.
I will check whether the [gradle bndtools] plugin allows a better workflow that is also compatible with Eclipse's PDE.

Reply:
I created a solution that does the following (based on the plugins of [buildmonkey](https://github.com/MONKEY-WORKS/BuildMonkey)):
* Convert the eclipse home directory (which is a p2 repository) to the maven repository structure
* Add this maven repository as local repository to gradle
* parse the MANIFEST.MF for required bundles
* add the required bundles to the set of compile dependencies
* resolve recursive dependencies to other bundles

Does this behaviour match with your expectations? It works in eclipse, but you have to trigger the p2->maven conversion once after the project is imported.